hw/xbox/mcpx_apu.c

```diff
@@ -473,6 +491,87 @@ static void ep_scratch_rw(void *opaque,
                       ptr, addr, len, dir);
 }
 
+static void circular_scatter_gather_rw(MCPXAPUState *d,
+                                       hwaddr sge_base, unsigned int max_sge,
+                                       uint8_t *ptr,
+                                       uint32_t base, uint32_t end,
+                                       uint32_t *cur,
```

Comment:
nit: This is kind-of ugly. Maybe just make this function return a `uint32_t` with the updated address?

*(On the other hand: this is probably closer to what the hardware does - updating register while writes are happening)*